packages/modules/src/modules/epics/ContributionsEpicChoiceCards.tsx

```diff
@@ -71,27 +61,62 @@ const contributionType: ContributionType = {
     },
 };
 
+// ContributionsEpicChoiceCards - exported component
+// -------------------------------------------
+export interface ChoiceCardSelection {
+    frequency: ContributionFrequency;
+    amount: number | 'other';
+}
+
+interface EpicChoiceCardProps {
+    selection?: ChoiceCardSelection;
+    setSelectionsCallback: (choiceCardSelection: ChoiceCardSelection) => void;
+    submitComponentEvent?: (event: OphanComponentEvent) => void;
+    currencySymbol: string;
+    amounts?: ContributionAmounts;
+    amountsTestName?: string;
+    amountsVariantName?: string;
+}
+
 export const ContributionsEpicChoiceCards: React.FC<EpicChoiceCardProps> = ({
-    amounts,
     selection,
     setSelectionsCallback,
-    countryCode,
     submitComponentEvent,
+    currencySymbol,
+    amounts,
+    amountsTestName = 'test_undefined',
+    amountsVariantName = 'variant_undefined',
 }: EpicChoiceCardProps) => {
-    const currencySymbol = getLocalCurrencySymbol(countryCode);
-    const countryGroupId = countryCodeToCountryGroupId(countryCode || 'GBPCountries');
-    const countryAmountsData = amounts[countryGroupId];
+    if (!selection || !amounts) {
+        return <></>;
+    }
 
-    // This is the bit we need to change for amounts testing
-    // Ideally we'll fix so the component is pre-supplied with the correct variant
-    const amountsTestVariant = countryAmountsData.control;
+    const abTestName = `${amountsTestName}__${amountsVariantName}`;
+
+    const [hasBeenSeen, setNode] = useHasBeenSeen({ threshold: 0 }, true) as HasBeenSeen;
+
+    useEffect(() => {
+        if (hasBeenSeen) {
+            // For ophan
+            if (submitComponentEvent) {
+                submitComponentEvent({
+                    component: {
+                        componentType: 'ACQUISITIONS_OTHER',
+                        id: abTestName,
+                    },
+                    action: 'VIEW',
+                });
+            }
+        }
+    }, [hasBeenSeen, submitComponentEvent]);
 
     const trackClick = (type: 'amount' | 'frequency'): void => {
         if (submitComponentEvent) {
             submitComponentEvent({
                 component: {
                     componentType: 'ACQUISITIONS_OTHER',
-                    id: `contributions-epic-choice-cards-change-${type}`,
+                    // id: `contributions-epic-choice-cards-change-${type}`,
+                    id: `${abTestName}-change-${type}`,
```

Comment:
I think it's best to keep this `id` as it was. The new AB test tracking will also be on the same pageview record

Reply:
Okay

Reply:
Fixed